### ISSUE TITLE
Add space-charge effects in Cheetah

### ISSUE THREAD

**Issue by RemiLehe** [desy-ml/cheetah#137]
It would be valuable to add space-charge effects in Cheetah.

This could be done by adding space-charge to existing elements by using the standard technique of sub-stepping through an element (used in other accelerator codes). For instance, a drift without space-charge
```
Drift(length=L)
```
would be replaced (here with 3 substeps) by
```
Segment(
    elements=[
        Drift(length=L/6),
       SpaceChargeKick(length=L/3),
       Drift(length=L/3),
       SpaceChargeKick(length=L/3),
       Drift(length=L/3),
       SpaceChargeKick(length=L/3),
       Drift(length=L/6)
  ]
)
```

One could use the integrated Green function method for `SpaceChargeKick`, described here:
https://journals.aps.org/prab/abstract/10.1103/PhysRevSTAB.9.044204

This could then be benchmarked with examples from https://inspirehep.net/literature/1690374, as well as the example from the section "Free Expansion of a Cold Uniform Bunch" in https://inspirehep.net/files/178e98af93e0aaf3838a175835781aa0

**Comment by ax3l** [comment]
Small comment: I'll take a look at  #142 :rocket:. Just for the above, technically, the space-charge kick is modeled like a thin element (no `lenghts=0`).

**Comment by RemiLehe** [comment]
Yes, we are aware that the space charge kick does not add length to the element (see for instance [this comment](https://github.com/desy-ml/cheetah/pull/142/files#r1590642938)). 
In `SpaceChargeKick(length=L/3)`, `length` is the equivalent of [`slice_ds` in ImpactX](https://github.com/ECP-WarpX/impactx/blob/development/src/particles/spacecharge/GatherAndPush.cpp#L25).

As [suggested by @jank324](https://github.com/desy-ml/cheetah/pull/142/files#r1590642938), the above `length` argument should probably be renamed (e.g. `effec_length`), maybe as part of a [follow-up PR](https://github.com/desy-ml/cheetah/issues/160) to #142 .

**Comment by ax3l** [comment]
Yes. In your link, it's not fully equivalent to our ds in ImpactX because we do not model space charge kicks conceptionally as _elements_. I stead, we slice elements (slice_ds=ds/N) and add collective effects between slices.

**Comment by jank324** [comment]
Closing this because #142 was merged a while ago.
